ArjunaJTA/jta/classes/com/arjuna/ats/internal/jta/transaction/arjunacore/jca/TransactionImporterImple.java

```diff
@@ -230,7 +230,13 @@ public Set<Xid> getInflightXids(String parentNodeName) {
 	 * @param timeout
 	 * @return
 	 */
-	private TransactionImportResult addImportedTransaction(TransactionImple recoveredTransaction, Xid mapKey, Xid xid, int timeout) {
+	private TransactionImportResult addImportedTransaction(TransactionImple recoveredTransaction, Xid mapKey, Xid xid, int timeout) throws XAException {
+		if (!TxControl.isEnabled()) {
+			// TODO needs looking at for the right code and so on and maybe it's different for recover than it is when importing a new transaction
+			// 	Consider this task as part of JBTM-3893
```

Comment:
Is it worth backing this out if it's not solved in JBTM-3894? Same for the other TODO in ArjunaJTS/jtax/classes/com/arjuna/ats/internal/jta/transaction/jts/jca/TransactionImporterImple.java?